File: Miscellaneous/Golang/169. Majority Element.go
package main

import "fmt"

func main() {

	arr := []int{2, 2, 1, 1, 1, 2, 2}
	result := majorityElement(arr)
	fmt.Println(result)
}

func majorityElement(arr []int) int {
	res, times := 0, 0

	for i := 0; i < len(arr); i++ {
		if times == 0 {
			res = arr[i]
		}
		if res == arr[i] {
			times++
		} else {
			times--
		}
	}

	return res
}

File: Miscellaneous/Python/169. Majority Element.py
from re import I


class Solution:
   def majorityElement(self, arr):
        res, times = 0, 0

        for num in arr:
            if times == 0:
                res = num
            elif res == num:
                times += 1
            else:
                times -= 1
        
        return res

obj = Solution()
arr = [2,2,1,1,1,2,2]
ans = obj.majorityElement(arr)
print(ans)